File: 3R/2010_04_SwitchingLights.cpp
//看看 sqrt後 和之前的是不是一樣
#include <iostream>
#include <cmath>

using namespace std;

int main(){
    int cases;
    cin >> cases;

    while(cases--){
        long m, n;
        cin >> m;
        n = (long)sqrt(m);

        cout << (n*n==m? "yes":"no") << endl;
    }

}

File: 3R/2010_04_SwitchingLights.java
//看看 sqrt後 和之前的是不是一樣
import java.util.*;

class Main{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
        int cases = sc.nextInt();

        while(cases-->0){
            long m = sc.nextLong();
            long n = (long)Math.sqrt(m);

            System.out.println(n*n==m? "yes":"no");
        }
	}
}